grid_map_visualization/src/GridMapVisualization.cpp

```diff
@@ -6,165 +6,170 @@
  *	 Institute: ETH Zurich, ANYbotics
  */
 
-#include "grid_map_visualization/GridMapVisualization.hpp"
 #include <grid_map_core/GridMap.hpp>
 #include <grid_map_ros/GridMapRosConverter.hpp>
 
-using namespace std;
-using namespace ros;
+#include <string>
+#include <vector>
+#include <memory>
+#include <chrono>
+#include <unordered_set>
+
+#include "grid_map_visualization/GridMapVisualization.hpp"
 
-namespace grid_map_visualization {
+namespace grid_map_visualization
+{
 
-GridMapVisualization::GridMapVisualization(ros::NodeHandle& nodeHandle,
-                                           const std::string& parameterName)
-    : nodeHandle_(nodeHandle),
-      visualizationsParameter_(parameterName),
-      factory_(nodeHandle_),
-      isSubscribed_(false)
+GridMapVisualization::GridMapVisualization(const std::string & parameterName)
+: visualizationsParameter_(parameterName)
+  // isSubscribed_(false)
 {
-  ROS_INFO("Grid map visualization node started.");
+  nodeHandle_ = std::make_shared<rclcpp::Node>("grid_map_visualization");
+  factory_ = std::make_shared<VisualizationFactory>(nodeHandle_);
+
+  RCLCPP_INFO(nodeHandle_->get_logger(), "Grid map visualization node started.");
   readParameters();
-  activityCheckTimer_ = nodeHandle_.createTimer(activityCheckDuration_,
-                                                &GridMapVisualization::updateSubscriptionCallback,
-                                                this);
+
+  activityCheckTimer_ = nodeHandle_->create_wall_timer(
+    std::chrono::duration<double>(1.0 / activityCheckRate_),
+    std::bind(&GridMapVisualization::updateSubscriptionCallback, this));
   initialize();
 }
 
 GridMapVisualization::~GridMapVisualization()
 {
 }
 
+rclcpp::Node::SharedPtr
+GridMapVisualization::get_node_ptr()
+{
+  return nodeHandle_;
+}
+
 bool GridMapVisualization::readParameters()
 {
-  nodeHandle_.param("grid_map_topic", mapTopic_, string("/grid_map"));
+  nodeHandle_->declare_parameter("grid_map_topic", std::string("/grid_map"));
```

Comment:
Please rename to `node`, nodehandles dont exist anymore and this won't age well

Reply:
sure.